include/RAJA/pattern/nested.hpp

```diff
@@ -0,0 +1,318 @@
+#ifndef RAJA_pattern_nested_HPP
+#define RAJA_pattern_nested_HPP
+
+#include "RAJA/RAJA.hpp"
+#include "RAJA/config.hpp"
+#include "RAJA/util/defines.hpp"
+#include "RAJA/util/types.hpp"
+
+// #include "RAJA/external/metal.hpp"
+#include "camp/camp.hpp"
+#include "camp/tuple.hpp"
+
+#include <iostream>
+#include <type_traits>
+
+namespace RAJA
+{
+namespace nested
+{
+
+namespace internal
+{
+
+template <typename T>
+using remove_all_t =
+    typename std::remove_cv<typename std::remove_reference<T>::type>::type;
```

Comment:
This could be used in many, many other places. Maybe a move to `camp` makes more sense. I can see users wanting this. It's a less-specialized `decay` :)